lib/active_job/queue_adapters/amazon_sqs_adapter.rb

```diff
@@ -27,11 +27,8 @@ def _enqueue(job, body = nil, send_message_opts = {})
         send_message_opts[:message_attributes] = message_attributes(job)
 
         if Aws::Rails::SqsActiveJob.fifo?(queue_url)
-          # job_id is unique per initialization of job
-          # Remove it from message dup id to ensure run-once behavior
-          # with ActiveJob retries
           send_message_opts[:message_deduplication_id] =
-            Digest::SHA256.hexdigest(Aws::Json.dump(body.except('job_id')))
+            Digest::SHA256.hexdigest(Aws::Json.dump(deduplication_body(job, body)))
```

Comment:
A general thought - rather than configuring the keys to either include or exclude - would it make sense to allow the user to configure a lambda/proc that would be called? That would allow full customization of the deduplication_id.  (But may be more complex for most use cases....)

Reply:
This is possibly overkill ..